messaging/Controllers/BundlesController.cs

```diff
@@ -31,15 +31,31 @@ public BundlesController(ApplicationDbContext context, IServiceProvider services
         public async Task<ActionResult<Bundle>> GetOutgoingMessageItems(string jurisdictionId, DateTime _since = default(DateTime))
         {
             // TODO only allow the since param in development
-            var messageTasks = _context.OutgoingMessageItems.Where(message => message.CreatedDate >= _since && message.JurisdictionId == jurisdictionId).ToList()
-                                                            .Select(message => System.Threading.Tasks.Task.Run(() => BaseMessage.Parse(message.Message, true)));
+            // if _since is the default value, then apply the retrieved at logic
+            System.Collections.Generic.IEnumerable<System.Threading.Tasks.Task<VRDR.BaseMessage>> messageTasks;
+            if (_since == default(DateTime))
+            {
+                // get all messages that have not yet been retrieved 
+                messageTasks = _context.OutgoingMessageItems.Where(message => message.RetrievedAt == null && message.JurisdictionId == jurisdictionId).ToList()
+                                                            .Select(message => System.Threading.Tasks.Task.Run(() => BaseMessage.Parse(message.Message, true))); 
+            }
+            else
+            {
+                messageTasks = _context.OutgoingMessageItems.Where(message => message.CreatedDate >= _since && message.JurisdictionId == jurisdictionId).ToList()

```

Comment:
I'm not sure on this one ... Robert wrote this code. My only guess so far is it is related to the Parsing so that it's still an IEnumerable. But I could see this causing an issue, I can flag it with a comment but don't have an immediate explanation